Added SetupHelp and GitProcess markdown files into Documentation (#100)
tumcms/Open-Infra-Platform: Documentation/markdown/GitProcess.md

> # Git Workflow

Reviewer: Suggestion for the structure of the document: Maybe separate the GitProcess.md into two parts: 
(1) GitSetup (which is once-off): Prerequisites, Fork
(2) GitWorkflow (which is to explain how to work with Git all the time):
- Updating from remote
-  Creating a branch (move this here from first part that is currently under "Submitting changes" and include branch naming conventions here
- Submit changes
- Pull request

(In this order as it is then the same as the actual workflow and especially so that you have latest version of tumcms development OIP before starting to make any local changes.)

Author: Done

	[More information about updating your fork](#Updating). 

	As a result you will have two remote repositories: 

		- Your **origin** repository (https://github.com/<your_username>/Open-Infra-Platform).

		- **tumcms** repository (https://github.com/tumcms/Open-Infra-Platform).

## Submit changes 

1. Before creating any changes, create new branch: 
 
	- Open OpenInfraPlatform folder and select *Git Bash Here*. 
	
	- You also can open **Git Bash** and write out in the command line:
	 > cd "Path to OpenInfraPlatform" (e.g  *cd  "C:\dev\Open-Infra-Platform"*).

	- In the command line write out:
	 > git checkout -b <branchname> (e.g. *git checkout -b Fixing_Bugs*).

2. After creating some changes, *add* these changes:

	- In the command line of **Git Bash** write out:
	 > git add <filename> 

	This will add changes of one file to a commit. 

	- **OR** write out:
	 > git add * 
	 
	This will add all changes you have made when working to a commit. 

3. *Commit* changes:

	- In the command line write out:
	 > git commit -m "Commit message" (e.g. *git commit -m "Fixed Bugs"*). 
	 
	**NOTE:** Please be descriptive in your commit message as to what changes are involved in this commit.

> 4. *Push* changes to remote repository: 

Author: Done

# Git-Process

***
## GitSetup

### Prerequisites 

Before you can proceed to work with Git, there are several steps, which have to be done before Installation: 

1. Installation of **Git** - find [here](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
2. Creation of **GitHub account** - find [here](https://github.com/)

### Fork the repository

1. Go to OpenInfraPlatform on [GitHub](https://github.com/tumcms/Open-Infra-Platform).

2. Login to your account and *Fork* OpenInfraPlatform to your account. Now there is https://github.com/<your_username>/Open-Infra-Platform. This is your **origin** repository. 

> 3. *Clone* your repository to local machine.

Reviewer 2: Typo

	- Open your File Explorer and select the folder, where you want to clone OpenInfraPlatform (e.g. C:\dev). Right mouse click in this folder and select *Git Bash Here*.

	- In the window, which you have opened, in the command line write out:

> 	 > git clone https://github.com/<your_username>/Open-Infra-Platform.git

Reviewer 2: This can also be copy-pasted directly from github after forking a repository

![image](https://user-images.githubusercontent.com/12066490/81430711-2b71fd00-9160-11ea-9c66-17d91c0a90b4.png)

Author: Done

1. Add all updates to your local repository: 

	- In **Git Bash**, which is working in OpenInfraPlatform folder, in the command line write out:
	 > git pull tumcms development

	*tumcms* is the name you gave to the OpenInfraPlatform official repository remote (https://github.com/tumcms/Open-Infra-Platform) when [adding remote](#Remote). 
	
	*development* is the name of the main branch in this repository. 

2. Update your **origin** repository:

	- In the command line write out:
	 > git push origin development 

	*origin* is the name of your remote repository (https://github.com/<your_username>/Open-Infra-Platform). 
	
	*development* is the name of the main branch in your remote repository. 

Now you have both your repositories (local and remote) updated. 

### Creating a branch

Before creating any changes, create new branch: 
 
- Open OpenInfraPlatform folder and select *Git Bash Here*. 
	
- You also can open **Git Bash** and write out in the command line:
 > cd "Path to OpenInfraPlatform" (e.g  *cd  "C:\dev\Open-Infra-Platform"*).

- In the command line write out:
 > git checkout -b <branchname> (e.g. *git checkout -b Fixing_Bugs*).

### Branch naming convention

> ***

Reviewer 2: Still to do?

> # Git Procedures 

Reviewer 3: Git Workflow

Author: Corrected

> Here you will find description about Git procedures you need working with OpenInfraPlatform

Reviewer 3: Add a link to / include the top figure from [here](https://www.c-sharpcorner.com/article/git-and-github-version-control-local-and-remote-repository/).

Author: Included Git_Workflow.png

# Git Workflow

![](./fig/Git_Workflow.png)

Here you will find description about Git Workflow you need working with OpenInfraPlatform

## 1. Clone the repository

1.1. Go to OpenInfraPlatform on [GitHub](https://github.com/tumcms/Open-Infra-Platform).

1.2. Login to your account and *Fork* OpenInfraPlatform to your account.

1.3. Select green button *Clone or download* and copy url, which will shows up.

1.4. Open your File Explorer and select the folder, where you want to clone the OpenInfraPlatform (e.g. C:\dev). Right mouse click in this folder and select *Git Bash Here*.

> 1.5. In the window, which you have opened, in the command line write down *git clone* and paste link you have just copied. This will start a download.

Reviewer 3: Write out the command *git clone https://github.com/<your_username>/Open-Infra-Platform.git*

Author: Done

# Git Workflow

![](./fig/Git_Workflow.png)

Here you will find description about Git Workflow you need working with OpenInfraPlatform

> ## 1. Clone the repository

Reviewer 3: No need for section headings' numbering in Markdown.

Author: Corrected

1. Installation of **Git** - 
2. Creation of **GitHub account** - 

## Fork the repository

1. Go to OpenInfraPlatform on [GitHub](https://github.com/tumcms/Open-Infra-Platform).

2. Login to your account and *Fork* OpenInfraPlatform to your account. Now there is https://github.com/<your_username>/Open-Infra-Platform. 

3. *Clone* your repository to local mashine.

	- Open your File Explorer and select the folder, where you want to clone the OpenInfraPlatform (e.g. C:\dev). Right mouse click in this folder and select *Git Bash Here*.

	- In the window, which you have opened, in the command line write out *git clone https://github.com/<your_username>/Open-Infra-Platform.git* . This will start a download.

4. Add remote: git add 

## Submit changes 

branching !!

(Commit + push)

##Pull Request 

(Pull Request)

## Updating from main repository 

> (Fetch all)...

Reviewer 3: This needs to be addressed before merging.

1. Go to OpenInfraPlatform on [GitHub](https://github.com/tumcms/Open-Infra-Platform).

2. Login to your account and *Fork* OpenInfraPlatform to your account. Now there is https://github.com/<your_username>/Open-Infra-Platform. This is your **origin** repository. 

3. *Clone* your repository to local machine.

	- Open your File Explorer and select the folder, where you want to clone OpenInfraPlatform (e.g. C:\dev). Right mouse click in this folder and select *Git Bash Here*.

	- In the window, which you have opened, in the command line write out *git clone https://github.com/<your_username>/Open-Infra-Platform.git* . This will start a download.

> 4. Add remote:  In the command line write out *git remote add tumcms https://github.com/tumcms/Open-Infra-Platform.git*. Adding a remote repository gives you an opportunity to *pull* changes from the official repository. While working with OpenInfraPlatform, official repository is https://github.com/tumcms/Open-Infra-Platform.git. For this reason, it is desired to name this repository **tumcms**, to distinguish it from your **origin** repository.  More information about updating your repositories find [here](./GitProcess.md#Updating from main repository). As a result you will have two remote repositories: 

Reviewer 3: ```suggestion
4. Add remote:  In the command line write out *git remote add tumcms https://github.com/tumcms/Open-Infra-Platform.git*. 

Adding a remote repository gives you an opportunity to *fetch* all changes from the official repository. While working with OpenInfraPlatform, official repository is https://github.com/tumcms/Open-Infra-Platform. For this reason, it is desired to name this remote **tumcms**, to distinguish it from your **origin** remote repository.  

[More information about updating your fork.](./GitProcess.md#Updating from main repository). 

As a result you will have two remote repositories: 
```

	- Alternatively, you can select *Pull requests*. There you can manually create *New pull request*. This will open a new page, where you will be able to select, which branch you want to pull in what another branch. As your compare changes not only between to different *branches*, but also between different *forks*, you should choose an option to *compare across forks*. As a result, you will have four selections possible, instead of two:
		
		- For *base repository* select *tumcms/Open-Infra-Platform*.

		- For *base* select *development*.

		- For *head repository* select *<your_username>/Open-Infra-Platform*.

		- For *compare* select *<branchname>* (e.g. Fixing_Bugs).

3. On the new page you have an option to comment all changes, you have made. Please be descriptive commenting all changes (include names of all files you have changed; describe main changes you have made to these files). 

4. Assign *reviewers* of your changes.

5. Assign *Labels* to your *pull request*. (e.g. bug)

6. Select *Create pull request*

## Updating from main repository 

(Fetch all)...

> ## Branch naming convention

Reviewer 3: Still open.

Author: Done

3. *Clone* your repository to local mashine.

	- Open your File Explorer and select the folder, where you want to clone OpenInfraPlatform (e.g. C:\dev). Right mouse click in this folder and select *Git Bash Here*.

	- In the window, which you have opened, in the command line write out *git clone https://github.com/<your_username>/Open-Infra-Platform.git* . This will start a download.

> 4. Add remote: *git remote add tumcms https://github.com/tumcms/Open-Infra-Platform.git*

Reviewer 3: Please add another sentence or two. Why "tumcms" as name, what does this command do, perhaps with a forward reference to the "Updating from main repo".

Author: Done

## Submit changes 

1. Before creating any changes, create new branch: 
 
	- Open OpenInfraPlatform folder and select *Git Bash Here*. You also can open **Git Bash** and write out in the command line *cd "<Path to OpenInfraPlatform>"* (e.g  *cd  "C:\dev\Open-Infra-Platform"*)

> 	- In the command line write out *git checkout -b <branchname>* (e.g *git checkout -b Fixing_Bugs*)

Reviewer 3: Please foresee another section "Branch naming convention". It can be left empty for now.

Author: Done

2. After creating some changes, *add* these changes:

> 	- In the command line of **Git Bash** write out *git add <filename>* to add changes to one file. 

Reviewer 3: ... to add changes of one file to a commit. Try and be as atomic as possible in commit contents.

Author: Corrected

	- **OR** write out -git add *- to add all changes you have made while working. 

3. *Commit* changes:

> 	- In the command line write out *git commit -m "Commit message"* (e.g. *git commit -m "Fixed Bugs"*)

Reviewer 3: add sth inthe way of: Please be descriptive in your commit message as to what changes are involved in this commit.

Author: Done

4.*Push* changes to remote repository: 

> 	- In the command line write out *git push*

Reviewer 3: "git push origin"

Reviewer 3: Also, add a comment, in sort of "Now your changes are online on GitHub in your repository.

Author: Done

5. Assign *Labels* to your *pull request*. (e.g. bug)

6. Select *Create pull request*.

## <a name="Updating"></a> Updating from main repository 

Once you have *added a remote*, you can update your local repository with all changes made in the official repository. **It is highly recommended to update your local repository every time you start working with it. Otherwise, you will not have access to recent changes**. 

1. Add all updates to your local repository: 

> 	- In **Git Bash**, which is working in OpenInfraPlatform folder, in the command line write out *git pull tumcms development*.  

Reviewer 3: ```suggestion
	- In **Git Bash**, which is working in OpenInfraPlatform folder, in the command line write out:
	 > git pull tumcms development
```

Reviewer 3: Please reformat all bash-commands to be on a new line starting with ">".

Author: Done

> 	*tumcms* is a name you gave to OpenInfraPlatform official repository (https://github.com/tumcms/Open-Infra-Platform) while [adding remote](#Remote). 

Reviewer 3: ```suggestion
	*tumcms* is the name you gave to the OpenInfraPlatform official repository remote (https://github.com/tumcms/Open-Infra-Platform) when [adding remote](#Remote). 
```

Reviewer 3: Apply similarly to other lines as well.

Author: Done
